source/MissionPanel.cpp

```diff
@@ -971,22 +971,48 @@ void MissionPanel::Accept(bool force)
 	if(availableIt == available.end() && !available.empty())
 		--availableIt;
 
+	cycleInvolvedIndex = 0;
+
 	// Check if any other jobs are available with the same destination. Prefer
 	// jobs that also have the same destination planet.
 	if(toAccept.Destination())
 	{
+		const list<Mission>::const_iterator startHere = availableIt;
 		const Planet *planet = toAccept.Destination();
 		const System *system = planet->GetSystem();
-		for(auto it = available.begin(); it != available.end(); ++it)
+		bool stillLooking = true;
+		for(auto it = startHere; it != available.end(); ++it)
 			if(it->Destination() && it->Destination()->IsInSystem(system))
 			{
-				availableIt = it;
 				if(it->Destination() == planet)
-					break;
+				{
+					availableIt = it;
+					return;
+				}
+				else if(stillLooking)
+				{
+					stillLooking = false;
+					availableIt = it;
+				}
 			}
+		for(auto it = startHere; it != available.begin(); )
```

Comment:
The `begin` iterator points to the first element, so you want to consider that.
Use the `rend` (reverse-end) iterator instead.
https://en.cppreference.com/w/cpp/iterator/reverse_iterator
It might be better to create a reverse iterator and use that for this.
```suggestion
		for(auto it = reverse_iterator(startHere), ++it; it != available.rend(); ++it)
```
(I haven't checked if that'll work but it seems like something that should.)
You should probably add something like `if(available.empty()) return;` at line 975 (after setting `cycleInvolvedIndex` to 0 and before looking for a new job to select. There's no point checking if there are no jobs to consider and ensuring taht the list has at least one element at this point ensures that incrementing the iterator does not push it beyond the `rend` iterator, because it will not start at `rend` as `startHere` will not refer to anything earlier than `begin`, and since there are items in the list, `end` is not the same as `rend` (I think.)

Reply:
If you check where the decrement is done in the reverse loop, it's before the value is used, so checked values range from startHere-1 to begin() (startHere was covered in the forward loop) - begin() is handled normally by the loop.

I don't think the reverse_iterator will work, because we want an iterator out for availableIt.  Also, I could not find a way to convert an iterator to a reverse_iterator (make_reverse_iterator shows up in c++14, which we aren't using yet?).

Sure, I can add a check for the list being empty.  It wasn't in the original, and the logic works if the list is empty, so I hadn't bothered.

Reply:
It looks like it'll need to be:
```suggestion
		for(auto it = reverse_iterator<decltype(startHere)>(startHere); it != available.rend(); ++it)
```
Which removes the need to decrement the iterator at the beginning of each loop step.

I think:
```c++
		for(auto it = make_reverse_iterator(startHere); it != available.rend(); ++it)
```
Would be best, but we only C++11 right now.

Though, using a reverse iterator here means we'd have to convert it back to assign it to availableIt.
That can be done with reverse_iterator::base().
`availableIt = it.base() - 1;`

This becomes a bit more complicated with the use of the lambda, though, so maybe not.